MvvmHelpers/Commands/CommandUtils.cs

```diff
@@ -0,0 +1,40 @@
+﻿using MvvmHelpers.Exceptions;
+using System;
+using System.Reflection;
+
+namespace MvvmHelpers.Commands
+{
+	internal static class CommandUtils
+	{
+		internal static bool IsValidCommandParameter<T>(object o)
+		{
+			bool valid;
+			if (o != null)
+			{
+				// The parameter isn't null, so we don't have to worry whether null is a valid option
+				valid = o is T;
+
+				if (!valid)
+					throw new InvalidCommandParameterException(typeof(T), null);
```

Comment:
Passing in `null` to `new InvalidCommandParameterException(Type expectedType, Type actualType)` creates a confusing error message:

```
Unhandled Exception: MvvmHelpers.Exceptions.InvalidCommandParameterException: Invalid type for parameter. Expected Type: System.Int32, but received Type: 
   at MvvmHelpers.Commands.CommandUtils.IsValidCommandParameter[T](Object o)
```

Since you want to let the user know that they incorrectly passed in a `null`, you could create  another constructor for `InvalidCommandParameterException`:

```csharp
public InvalidCommandParameterException(Type expectedType) : base("Invalid type for parameter. Expected Type: {expectedType}, but received Type: null")
{

} 
```